protocol/xmrmaker/net.go

```diff
@@ -51,8 +51,10 @@ func (inst *Instance) initiate(
 		}
 	}
 
-	// checks passed, delete offer for now
-	inst.offerManager.DeleteOffer(offer.ID)
+	// checks passed, delete the offer for now
+	if err = inst.offerManager.DeleteOffer(offer.ID); err != nil {
+		return nil, err
+	}
```

Comment:
`DeleteOffer` will only return serious errors. Deleting an offer that doesn't exist is error free. I think that behavior is much improved over not returning any errors, but I can see an argument even the new behavior is not homogeneous with the `GetOffer` behavior that returns an error if the offer does not exist. I'm okay with it, but I'm open to alternative suggestions.